src/TypeScript/Rules/ActiveJavaScriptRulesProvider.cs

```diff
@@ -0,0 +1,78 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Linq;
+using SonarLint.VisualStudio.TypeScript.EslintBridgeClient.Contract;
+
+namespace SonarLint.VisualStudio.TypeScript.Rules
+{
+    interface IActiveJavaScriptRulesProvider
+    {
+        /// <summary>
+        /// Returns the eslint configuration for the currently active rules
+        /// </summary>
+        IEnumerable<Rule> Get();
+    }
+
+    [Export(typeof(IActiveJavaScriptRulesProvider))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class ActiveJavaScriptRulesProvider : IActiveJavaScriptRulesProvider
```

Comment:
This is a minimal implementation to allow us to get an end-to-end analysis working.

It works directly against the list of rule definitions and does some simple filtering for rules we know we don't want to run i.e. not active, hotspot.

I haven't bothered with any perf optimisations at this stage so the list is recalculated each time, even though it will currently return the same list.
That will change once we start implementing some of the TODOs.